lib/shared/config/cli-flags.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+var yargs = require('yargs');
+var camelCase = require('lodash.camelcase');
+
+function mergeToCliFlags(cliFlags, config, cliOptions) {
+  var argv = yargs(process.argv.slice(2)).argv;
+  var cfgFlags = config.flags || {};
+
+  var optNames = ['silent'];
+  optNames.filter(excludeArgsByUser).map(camelCase).forEach(copyConfig);
+
+  function excludeArgsByUser(name) {
+    return !(argv[name] != null || argv[cliOptions[name].alias] != null);
+  }
+
+  function copyConfig(name) {
```

Comment:
Instead of copying the config, you should be generating a new object and mapping onto that. (likely using `.reduce`)